lib/adbc_database.ex

```diff
@@ -59,6 +59,146 @@ defmodule Adbc.Database do
     end
   end
 
+  @doc """
+  Get a string type option of the database.
+  """
+  @spec get_string_option(pid(), atom() | String.t()) :: {:ok, String.t()} | {:error, String.t()}
+  def get_string_option(db, key) when is_pid(db) do
+    case GenServer.call(db, {:get_option, :string, to_string(key)}) do
+      {:ok, value} ->
+        {:ok, value}
+
+      {:error, reason} ->
+        {:error, error_to_exception(reason)}
+    end
+  end
+
+  @doc """
+  Get a binary (bytes) type option of the database.
+  """
+  @spec get_binary_option(pid(), atom() | String.t()) :: {:ok, binary()} | {:error, String.t()}
+  def get_binary_option(db, key) when is_pid(db) do
+    case GenServer.call(db, {:get_option, :bytes, to_string(key)}) do
+      {:ok, value} ->
+        {:ok, value}
+
+      {:error, reason} ->
+        {:error, error_to_exception(reason)}
+    end
+  end
+
+  @doc """
+  Get an integer type option of the database.
+  """
+  @spec get_integer_option(pid(), atom() | String.t()) :: {:ok, integer()} | {:error, String.t()}
+  def get_integer_option(db, key) when is_pid(db) do
+    case GenServer.call(db, {:get_option, :int, to_string(key)}) do
+      {:ok, value} ->
+        {:ok, value}
+
+      {:error, reason} ->
+        {:error, error_to_exception(reason)}
+    end
+  end
+
+  @doc """
+  Get a float type option of the database.
+  """
+  @spec get_float_option(pid(), atom() | String.t()) :: {:ok, float()} | {:error, String.t()}
+  def get_float_option(db, key) when is_pid(db) do
+    case GenServer.call(db, {:get_option, :double, to_string(key)}) do
+      {:ok, value} ->
+        {:ok, value}
+
+      {:error, reason} ->
+        {:error, error_to_exception(reason)}
+    end
+  end
+
+  @doc """
+  Set option for the database.
+
+  - If `value` is an atom or a string, then corresponding string option will be set.
+  - If `value` is a `{:byte, binary()}`-tuple, then corresponding binary option will be set.
```

Comment:
Here it says `:byte`, the docs says `:bytes`. :) But Maybe we should use `:binary`?